### PR TITLE
CTV-2835: Adding Comcast Support

### PR DESCRIPTION
**Pull request by louisachu**
### JIRA
https://truextech.atlassian.net/browse/CTV-2835

### Description
Ensure ads run on Comcast
localStorage does not exists in Comcast so need to use cookies instead. 

### Testing
run unit testing ono the added cookies support. 

### Commit Message Subject(s)
CTV-2835: Adding Comcast Support

### Additional Context
n/a

### Related PRs
OPTIONAL_ADD_PR_LINKS

### Checks
- [ ] Includes unit test coverage _(if applicable)_
- [ ] Includes functional test coverage _(at feature-level, if applicable)_
- [ ] Proposed commit messages follow [team conventions](https://github.com/socialvibe/adlabs-wiki/blob/develop/git/README.md#commits)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## v1.3.3
```
Comment (rblaa) [inline CHANGELOG.md]: looks like there are conflicts, you have to merge in the latest and bump the version, retag

File: src/focus_manager/txm_platform.js [hunk 2]
```diff
@@ -247,6 +248,9 @@ export class TXMPlatform {
             || /CrKey/.test(userAgent)) {
             configureForVizio();
 
+        } else if (/Linux/.test(userAgent) && /WPE/i.test(userAgent)) {
```
Comment (rblaa) [inline src/focus_manager/txm_platform.js]: I don't think we want the /i, we want exact case match
actually let's be more precise, and say it is its own word as well:
```
&& /\bWPE\b/i.test(userAgent)

File: src/utils/__tests__/app-storage-test.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { AppStorage } from "../app-storage";
+
+describe("AppStorage", () => {
+  const appStorage = new AppStorage();
+  test("AppStorage setItem", () => {
```
Comment (rblaa) [inline src/utils/__tests__/app-storage-test.js:5]: can you add a test that sets localStorage to be null and then verify that get/set a value works now with cookies

File: yarn.lock [hunk 4]
```diff
@@ -2,18 +2,30 @@
 # yarn lockfile v1
 
 
-"@babel/code-frame@^7.0.0", "@babel/code-frame@^7.12.13":
+"@babel/code-frame@^7.12.13":
```
Comment (rblaa) [inline yarn.lock]: get the latest merged yarn.lock file instead, there should be little to no changes, except perhaps for your jest update

File: package.json [hunk 5]
```diff
@@ -16,10 +18,14 @@
     "@babel/runtime": "^7.4.1",
     "babel-jest": "^25.1.0",
     "core-js": "3",
-    "jest": "^24.0.0"
+    "jest": "^27.0.6"
```
Comment (rblaa) [inline package.json]: you fail for both versions?
if so, let's revert since I don't want to risk breaking any test assumptions just yet

**Review by rblaa — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (rblaa) [reply CHANGELOG.md]: maybe don't tag yet until the PR is merged

File: src/utils/app-storage.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
+export class AppStorage {
+  // For some platform (e.g. Comcast), there is no local storage.
```
Comment (rblaa) [inline src/utils/app-storage.js]: let's put this as a regular "doc" comment above the class, e.g.
```
/**
 * ...
 */

File: src/utils/__tests__/app-storage-test.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    expect(appStorage.getItem("test1")).toBe("storage1");
+    appStorage.setItem("test2", JSON.stringify(["storage2, storage3"]));
+    var test2Result = JSON.parse(appStorage.getItem("test2"));
+    expect(test2Result).toStrictEqual(["storage2, storage3"]);
+    appStorage.removeItem("test1");
+    expect(appStorage.getItem("test1")).toBeNull();
+    appStorage.removeItem("test2");
+    expect(appStorage.getItem("test2")).toBeNull();
+
+    localStorage = null;
```
Comment (rblaa) [inline src/utils/__tests__/app-storage-test.js:17]: let's put this in its own test() method, e.g. test("AppStorage with null localStorage", ...);

File: src/utils/app-storage.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
+export class AppStorage {
+  /* For some platform (e.g. Comcast), there is no local storage.
```
Comment (rblaa) [inline src/utils/app-storage.js]: use this format, put above the class as per the usual practice
```
/**
 * For some...
 */

**Review by rblaa — CHANGES_REQUESTED** [review]

**Review by rblaa — APPROVED** [review]
